apps/fz_http/lib/fz_http/devices/device_setting.ex

```diff
@@ -0,0 +1,56 @@
+defmodule FzHttp.Devices.DeviceSetting do
+  @moduledoc """
+  Device setting parsed from either a Device or Map.
+  """
+  use Ecto.Schema
+
+  import Ecto.Changeset
+  import FzHttp.Devices, only: [decode: 1]
+
+  alias FzHttp.Devices.Device
+
+  @primary_key false
+  embedded_schema do
+    field :ip, :string
+    field :ip6, :string
+    field :user_id, :integer
+  end
+
+  def parse(%Device{} = device) do
+    %__MODULE__{
+      ip: decode(device.ipv4),
+      ip6: decode(device.ipv6),
+      user_id: device.user_id
+    }
+  end
+
+  def parse(device) when is_map(device) do
+    device =
+      device
+      |> Map.put(:ip, Map.get(device, :ipv4))
+      |> Map.put(:ip6, Map.get(device, :ipv6))
+
+    %__MODULE__{}
+    |> cast(device, [:ip, :ip6, :user_id])
+    |> validate_required(:user_id)
+    |> validate_ip_required()
+    |> validate_ip6_required()
+    |> apply_changes()
+  end
+
+  defp validate_ip_required(changeset) do
+    if Application.fetch_env!(:fz_http, :wireguard_ipv4_enabled) do
+      validate_required(changeset, :ip)
+    else
+      changeset
+    end
+  end
+
+  defp validate_ip6_required(changeset) do
+    if Application.fetch_env!(:fz_http, :wireguard_ipv6_enabled) do
+      validate_required(changeset, :ip6)
+    else
+      changeset
+    end
+  end
```

Comment:
What happens when there are existing records and config changes? (only ipv4 exists but ipv6 got turned on)

If we are not dealing with user inputs, we shouldn't really be validating anything.

Reply:
Yeah... I may just return maps from this. It seeemed like a bit of overkill, since both of these values are coming in from the db which has already been validated.